libusb/os/darwin_usb.h

```diff
@@ -160,15 +160,16 @@
 #define IO_OBJECT_NULL ((io_object_t) 0)
 #endif
 
+/* returns the current macos version in a format similar to the
+ * MAC_OS_X_VERSION_MIN_REQUIRED macro.
+ * Examples:
+ *   10.1.5 -> 100105
+ *   13.3.0 -> 130300
+ */
+uint32_t get_running_version(void);
+
 /* Testing availability */
-#ifndef __has_builtin
-  #define __has_builtin(x) 0  // Compatibility with non-clang compilers.
-#endif
-#if __has_builtin(__builtin_available)
-  #define HAS_CAPTURE_DEVICE() __builtin_available(macOS 10.10, *)
-#else
-  #define HAS_CAPTURE_DEVICE() 0
-#endif
+#define HAS_CAPTURE_DEVICE() (get_running_version() >= 101000)
```

Comment:
Since the implementation of `get_running_version` uses a `static`, it's not very thread-safe, and so this macro has changed from thread-safe to not.

Looks like your other commits remove 2 of the 3 uses of this macro, perhaps this macro should just die.

Reply:
Good point about the macro. Will remove it since it doesn't help with readability.